ddtrace/datadog/include/exporter.hpp

```diff
@@ -0,0 +1,163 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+#pragma once
+
+#include <array>
+#include <chrono>
+#include <string>
+#include <string_view>
+#include <memory>
+#include <vector>
+
+extern "C" {
+#include "datadog/profiling.h"
+};
+
+namespace Datadog {
+
+// Forward
+class Profile;
+
+class DdogProfExporter {
+  void add_tag(ddog_Vec_Tag &tags, std::string_view key, std::string_view val);
+
+public:
+  static constexpr std::string_view language = "python";
+  static constexpr std::string_view family = "python";
+  static constexpr std::string_view profiler_version = "1.8.0rc2_libdatadog";
+
+  DdogProfExporter(std::string_view env, std::string_view service, std::string_view version, std::string_view url);
+  ~DdogProfExporter();
+
+  ddog_prof_Exporter *ptr;
+};
+
+class Uploader {
+  std::string env;     // ex: staging / local / prod
+  std::string service; // service name (ex:prof-probe-native)
+  std::string version; // appended to tags (example: 1.2.1)
+  std::string url;     // host:port
+  std::string api_key; // Datadog api key
+  bool agentless; // Whether or not to actually use API key/intake
+
+  std::unique_ptr<DdogProfExporter> ddog_exporter;
+
+public:
+  Uploader(std::string_view _env = "prod",
+           std::string_view _service = "py_libdatadog",
+           std::string_view _version = "",
+           std::string_view _url = "http://localhost:8126");
+
+  bool upload(const Profile *profile);
+
+};
+
+class Profile {
+  bool is_valid = false;
+
+  std::vector<ddog_prof_Location> locations;
```

Comment:
I think we are using these mostly as standard arrays as we expect stable addressing.